src/attributes/diagnostics.md

```diff
@@ -492,6 +492,47 @@ error[E0277]: My Message for `ImportantTrait<i32>` implemented for `String`
    = note: Note 2
 ```
 
+### The `diagnostic::do_not_recommend` attribute
+
+The `#[diagnostic::on_unimplemented]` attribute is a hint to the compiler to not show a certain trait implementation as part of the error message.
```

Comment:
Can this explain a little more what error this is referring to, and what circumstances it will avoid showing an error?

Reply:
> Can this explain a little more what error this is referring to

Currently this is mostly relevant for E0277, although that might change in the future?

> and what circumstances it will avoid showing an error?

It never avoids showing an error, it just changes the content of the error message to not mention/consider certain trait implementations if somewhat reasonable possible.